app/test/unit/git/commit-test.ts

```diff
@@ -514,24 +516,75 @@ describe('git/commit', () => {
     })
   })
 
-  describe('createMergeCommit with a merge conflict', () => {
-    let repository: Repository
-    describe('with a merge conflict', () => {
+  describe('createMergeCommit', () => {
+    describe('with a simple merge conflict', () => {
+      let repository: Repository
+      describe('with a merge conflict', () => {
+        beforeEach(async () => {
+          repository = await setupConflictedRepo()
+        })
+        it('creates a merge commit', async () => {
+          const status = await getStatusOrThrow(repository)
+          const trackedFiles = status.workingDirectory.files.filter(
+            f => f.status.kind !== AppFileStatusKind.Untracked
+          )
+          const sha = await createMergeCommit(repository, trackedFiles)
+          const newStatus = await getStatusOrThrow(repository)
+          expect(sha).toHaveLength(7)
+          expect(newStatus.workingDirectory.files).toHaveLength(0)
+        })
+      })
+    })
+
+    describe('with a merge conflict and manual resolutions', () => {
+      let repository: Repository
       beforeEach(async () => {
-        repository = await setupConflictedRepo()
+        repository = await setupConflictedRepoWithMultipleFiles()
+      })
+      it('keeps files chosen to be added and commits', async () => {
+        const status = await getStatusOrThrow(repository)
+        const trackedFiles = status.workingDirectory.files.filter(
+          f => f.status.kind !== AppFileStatusKind.Untracked
+        )
+        const manualResolutions = new Map([
+          ['bar', ManualConflictResolutionKind.ours],
+        ])
+        const sha = await createMergeCommit(
+          repository,
+          trackedFiles,
+          manualResolutions
+        )
+        expect(await FSE.pathExists(path.join(repository.path, 'bar'))).toBe(
```

Comment:
:art: is it worth adding a test in for the "theirs" case here to see that the resolution is applied correctly and the file is removed from the working directory? It's not clear to me what the options were based on the setup in `setupConflictedRepoWithMultipleFiles`...